server.js

```diff
@@ -1,30 +1,142 @@
 import express from "express";
 import cors from "cors";
+import booksData from "./data/books.json";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+
+// App running on port 8080
 const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Welcome page
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Welcome to The Book Server. Enter a path provided in the documentation to get book data");
+});
+
+// get the whole booklist (and filter by title, author or id)
+app.get("/books", (req, res) => {
+  const { title, author, id } = req.query;
+
+  let filteredBooks = booksData;
+
+  if (title) {
+    filteredBooks = filteredBooks.filter((book) => {
+      return book.title.toLowerCase().includes(title.toLowerCase());
+    });
+  }
+
+  if (author) {
+    filteredBooks = filteredBooks.filter((book) => {
+      return book.authors.toLowerCase().includes(author.toLowerCase());
+    });
+  }
+
+  if (id) {
+    filteredBooks = filteredBooks.filter((book) => {
+      return book.bookID === Number(id);
+    });
+  }
+  // If there is more than 0 matches, this json is returned 
+  if (filteredBooks.length > 0) {
+    res.status(200).json({
+      success: true,
+      message: `Found ${filteredBooks.length} books`,
+      body: {
+        books: filteredBooks,
+      },
+    });
+  } else {  // if no matches found
+    res.status(404).json({
+      success: false,
+      message: "404 No matching books found",
+      body: {},
+    });
+  }
+});
+
+//Sorting function for authors in ascending order
+app.get("/books/authors", (req, res) => {
+  
+  const sorting = (arr, key, direction = 'asc') => {   
+    if (direction === 'asc') {
+      return arr.sort((a, b) => (a[key] > b[key]) ? 1 : (b[key] > a[key] ? -1 : 0))
+    } else {
+      return arr.sort((a, b) => (a[key] < b[key]) ? 1 : ((b[key] < a[key]) ? -1 : 0))
+    }
+  };
+  // passes the bookData and the key 'authors' to the sorting function and saves to variable.
+  const sortedByAuthorAsc = sorting(booksData, 'authors');
+  
+  if (sortedByAuthorAsc.length > 0) {
+    res.status(200).json({
+      success: true,
+      message: 'Books sorted by author',
```

Comment:
Maybe you could add to the message that it is in ascending order? Just to make it extra clear.